### PR TITLE
fribidi: update source

### LINKED ISSUE #32709: fribidi download source is gone
## Issue description
fribidi.org is no longer available and redirects to github; the download URL 404s

### Steps to reproduce
Try to build fribidi on 1a98fc40e9bbfe6bdf17930b5fa926b3f50e5d27 ; I notice that you can get lucky and get it from nix cache but it should be updated

## Technical details
Sadly just doing 

```
-  src = fetchurl {
-    url = "http://fribidi.org/download/${name}.tar.bz2";
-    sha256 = "13jsb5qadlhsaxkbrb49nqslmbh904vvzhsm5mm2ghmv29i2l8h8";
+  src = fetchFromGitHub {
+    owner = "fribidi";
+    repo = "fribidi";
+    rev = version;
+    sha256 = "10q5jfch5qzrj2w4fbkr086ank66plx8hp7ra9a01irj80pbk96d";
   };
```

didn't seem enough: it can't find `aclocal`, there's no configure file out of the box as it expects… I don't know why it's different from the previous tarball.

### PR DESCRIPTION
**Pull request by lukateras**
###### Motivation for this change

Fixes #32709.

/cc @Fuuzetsu

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by orivej** [comment]
Why did you add `? null` to `glib`?

**Comment by yegortimoshenko** [comment]
@orivej It's an optional dependency.

**Comment by orivej** [comment]
It looks like `fribidi` uses `glib` only for tests, and the only reason it enters closure is due to `lib/libfribidi.la` file:
```
# Libraries that this one depends upon.
dependency_libs=' -L/nix/store/4bxgkmmn720w0jikk5sryd9w03wrgyb8-glib-2.54.2/lib -L/nix/store/8xvn6rvlz70n3hwqfr8l5sn1p9l6sqwr-pcre-8.41/lib /nix/store/4bxgkmmn720w0jikk5sryd9w03wrgyb8-glib-2.54.2/lib/libglib-2.0.la /nix/store/8xvn6rvlz70n3hwqfr8l5sn1p9l6sqwr-pcre-8.41/lib/libpcre.la -lpthread'
```
I'll remove the dependency and add a comment saying that it is not needed.

**Comment by yegortimoshenko** [comment]
I see. Thanks!

File: pkgs/development/libraries/fribidi/default.nix [hunk 1]
```diff
@@ -1,22 +1,26 @@
...
-  hardeningDisable = [ "format" ];
+  nativeBuildInputs = [ autoreconfHook pkgconfig ];
+
+  # Configure script checks for glib, but it is only used for tests.
 
   outputs = [ "out" "devdoc" ];
 
   meta = with stdenv.lib; {
-    homepage = http://fribidi.org/;
+    homepage = https://github.com/fribidi/fribidi/;
```
Comment (jtojnar) [inline pkgs/development/libraries/fribidi/default.nix]: We could probably do without the trailing slash.

Comment (yegortimoshenko) [reply pkgs/development/libraries/fribidi/default.nix]: Fixed!

**Pull request merged**